### PR TITLE
Brainmed Update

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Jesus Christ it's done.

- [ ] Wiki Update

changes: 
  - experiment: "An entirely new medical system! Holy balls that's a lot of changes."
  - experiment: "Pain has been SUBSTANTIALLY changed. It now depends on total damage to all your limbs."
  - rscadd: "Traumas no longer happen due to oxyloss."
  - tweak: "Inaprovaline has been renamed to norepinephrine and viceversa."
  - tweak: "CPR has been fixed up a bit. It has a small chance to restart hearts and can work with or without mouth-to-mouth."
  - rscadd: "Humans no longer die when hitting 200 damage. Dionae, IPCs, grues and zombies are exceptions."
  - rscadd: "Humans now die when their brain dies. For the brain to die, it must not have oxygen for quite a while."
  - rscadd: "Suffocation damage is now a modifier to blood. If you don't have much oxygen, then you'll get the effects of low blood."
  - rscadd: "Toxins now directly target organs. They'll try to destroy the liver and kidneys first... then move onto other organs. Be careful."
  - rscadd: "All organs in the body now work towards making sure the brain lives. The liver filters toxins, the kidneys do the same, the lungs provide oxygen, the heart pumps blood and so on."
  - rscadd: "Humans now have two new stats: blood oxygenation and blood pressure, shown on sensors and scanners. Blood pressure depends on how well your heart is pumping your blood. Blood oxygenation depends on how well your blood is oxygenated."
  - rscadd: "Your heart can now stop when your pulse is too high. Your pulse depends on many factors - like pain or how much blood your brain wants. A heart in asystole cannot pump blood, meaning your brain will suffocate."
  - rscadd: "When your heart stops, you go into asystole, aka flatlining. This can be fixed with CPR. Make sure that you fix the causes first... or they'll just go into asystole again."
  - rscadd: "Norepinephrine stabilizes your heartrate. This is a VERY important fact, do NOT forget this. Norepinephrine will be your best tool!"
  - tweak: "Tricordazine recipe is now 1u epinephrine 1u dylovene."
  - tweak: "Adrenaline has been added. Its recipe is 1u norepinephrine 1u hyperzine 1u dexalinplus."
  - tweak: "Bicaridine recipe is now 1u norepinephrine 1u carbon."
  - tweak: "Inaprovaline recipe is now 1u atropine 0.1u phoron 1u adrenaline."
  - tweak: "Paroxetine recipe is now 1u mindbreaker 1u acetone 1u norepinephrine."
  - tweak: "Nightlife recipe is now 1u methylphenidate 1u synaptizine 1u nitroglycerin."
  - tweak: "RMT recipe is now 1u potassium 1u norepinephrine."
  - rscadd: "Ondansetron, a chemical that prevents vomiting and nausea, is made with 1u antihistamine 1u synaptizine 1u water."
  - rscadd: "Dexalin and dexalin+ no longer straight up heal oxyloss. Instead, they now fool your brain into thinking that it has 50% oxygen and 80% oxygen respectively."
  - rscadd: "Alkysine now only heals the brain if it's properly oxygenated."
  - rscadd: "Added a stomach organ. This controls reagent ingestion and food ingestion."
  - rscadd: "Added a stomach pump mode to sleepers to pump INGESTED reagents out of people's stomach."
  - tweak: "Peridaxon now heals brain damage, if it's not damaged much at all."
  - tweak: "Dylovene no longer outright heals toxin damage. Instead, it removes toxin reagents from the blood."
  - rscadd: "New side effects to most healing chems. Be careful."
  - rscadd: "Deltamivir now inflicts a fever. Be careful, going overboard with the dosage can give you a pretty bad one. Try not to go over 20 units at once."
  - rscadd: "Bicaridine now gives you muscle spasms."
  - rscadd: "Kelotane makes you drop your held items. Dermaline does the same, as it is now a subtype."
  - rscadd: "Throwing beakers now shatters them. Be careful."
  - rscdel: "Numbers have been deleted from scanners. Mostly. Stop using terms like brute and burn damage, boys."
  - rscadd: "Taking a good chunk of damage at once now produces adrenaline. Acts like a painkiller and raises your pulse. Might also restart your heart."
  - tweak: "Soulstones work only when someone's in asystole."
  - tweak: "Changed how getting organ damage works. Lasers now deal organ damage."
  - tweak: "Most organs can now take more punishment."
  - tweak: "The head is now harder to hit."
  - tweak: "The organ damage dealt now depends on the size of the organ in the area you hit."
  - tweak: "Surgery pain is MUCH higher if you're awake."
  - tweak: "Self surgery works again."
  - tweak: "Using a scalpel on necrotic organs now fixes the necrosis. However, it'll also lower the organ's max damage as well."

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!wip

**Comment by NonQueueingMatt** [comment]
!review

Good luck, reviewers.

File: code/__defines/chemistry.dm [hunk 1]
```diff
@@ -45,22 +45,41 @@
...
-#define CE_PACIFIED "pacified"
-
-// Chemistry lists.
-var/list/tachycardics  = list("coffee", "inaprovaline", "hyperzine", "nitroglycerin", "thirteenloko", "nicotine") // Increase heart rate.
-var/list/bradycardics  = list("neurotoxin", "cryoxadone", "clonexadone", "space_drugs", "stoxin")                 // Decrease heart rate.
-var/list/heartstopper  = list("potassium_chlorophoride", "zombie_powder") // This stops the heart.
-var/list/cheartstopper = list("potassium_chloride")                       // This stops the heart when overdose is met. -- c = conditional
+// Apply status effects
+#define CE_ALCOHOL      "alcohol"		// Liver filtering
+#define CE_ANTIEMETIC   "antiemetic"	// suppesses vomiting
```
Comment (Geevies) [inline code/__defines/chemistry.dm]:
```suggestion
#define CE_ANTIEMETIC   "antiemetic"	// suppresses vomiting
```

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 2]
```diff
@@ -444,3 +423,15 @@ This function restores all organs.
...
 		rads = rads * species.radiation_mod
 	..(rads)
+
+/mob/living/carbon/human/proc/get_shock()
+
+	if (!can_feel_pain())
+		return 0
+
+	var/traumatic_shock = getHalLoss()                 // Pain.
+	traumatic_shock -= chem_effects[CE_PAINKILLER] // TODO: check what is actually stored here.
```
Comment (Geevies) [inline code/modules/mob/living/carbon/human/human_damage.dm]: cursed todo

File: code/modules/mob/living/carbon/human/life.dm [hunk 3]
```diff
@@ -287,7 +287,7 @@
 	total_radiation = Clamp(total_radiation,0,100)
 
 	if (total_radiation)
-		//var/obj/item/organ/diona/nutrients/rad_organ = locate() in internal_organs
+		//var/obj/item/organ/internal/diona/nutrients/rad_organ = locate() in internal_organs
```
Comment (Geevies) [inline code/modules/mob/living/carbon/human/life.dm]: deadcode or future code?

File: config/example/config.txt [hunk 4]
```diff
@@ -520,7 +520,7 @@ CARGO_LOAD_ITEMS_FROM json
 
 
 # Forces the maploader to load the specified map path. Valid options: aurora, runtime
-FORCE_MAP aurora
+FORCE_MAP runtime
```
Comment (Geevies) [inline config/example/config.txt]:
```suggestion
FORCE_MAP aurora
```

File: html/changelogs/mattjadegeeves-brainmed.yml [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  - rscadd: "Your heart can now stop when your pulse is too high. Your pulse depends on many factors - like pain or how much blood your brain wants. A heart in asystole cannot pump blood, meaning your brain will suffocate."
+  - rscadd: "When your heart stops, you go into asystole, aka flatlining. This can be fixed with CPR. Make sure that you fix the causes first... or they'll just go into asystole again."
+  - rscadd: "Norepinephrine stabilizes your heartrate. This is a VERY important fact, do NOT forget this. Norepinephrine will be your best tool!"
+  - rscadd: "Dexalin and dexalin+ no longer straight up heal oxyloss. Instead, they now fool your brain into thinking that it has 50% oxygen and 80% oxygen respectively."
+  - rscadd: "Alkysine now only heals the brain if it's properly oxygenated."
+  - rscadd: "Added a stomach organ. This controls reagent ingestion and food ingestion."
+  - rscadd: "Sleepers in dialysis mode now work as a stomach pump."
```
Comment (Geevies) [inline html/changelogs/mattjadegeeves-brainmed.yml]:
```suggestion
```

Comment (Geevies) [inline html/changelogs/mattjadegeeves-brainmed.yml]: I don't think this was ever added, as it needed UI stuff, unless you added it.

File: code/game/machinery/adv_med.dm [hunk 6]
```diff
@@ -335,37 +335,76 @@
...
+			if(!brain || occupant.stat == DEAD || (occupant.status_flags & FAKEDEATH))
+				brain_result = 0
+			else if(occupant.stat != DEAD)
+				brain_result = round(max(0,(1 - brain.damage/brain.max_damage)*100))
+		else
+			brain_result = -1
+
+		switch(brain_result)
+			if(0)
+				brain_result = "<span class='bad'>none, patient is braindead</span>"
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/game/machinery/adv_med.dm]: This is not meant to be capitalized on purpose

File: code/game/objects/items/devices/scanners.dm [hunk 7]
```diff
@@ -23,210 +23,257 @@ BREATH ANALYZER
...
 	src.add_fingerprint(user)
 	return
 
 /proc/calcDamage(var/DMGValue)
 	switch(DMGValue)
-		if (0 to 1)
-			return "Healthy"
-		if (1 to 10)
-			return "Negligible"
+		if(0 to 1)
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/game/objects/items/devices/scanners.dm]: They look good imo

File: code/game/objects/items/devices/scanners.dm [hunk 8]
```diff
@@ -23,210 +23,257 @@ BREATH ANALYZER
...
-		BR = M.getBruteLoss()
+	var/mob/living/carbon/human/H = M
+
+	if(H.isSynthetic())
+		to_chat(user, "<span class='warning'>This scanner is designed for organic humanoid patients only.</span>")
+		return
+
+	. = list()
+	var/header = list()
+	var/b
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/game/objects/items/devices/scanners.dm]: it literally is b, as in a bold tag

**Comment by NonQueueingMatt** [comment]
Fixes #7570 
Fixes #7578 
Fixes #7590

**Pull request merged**